contracts/modules/SwapsExternal.sol

```diff
@@ -42,7 +42,14 @@ contract SwapsExternal is VaultController, SwapsUser {
 			require(msg.value == sourceTokenAmount, "sourceTokenAmount mismatch");
 			wrbtcToken.deposit.value(sourceTokenAmount)();
 		} else {
+			IERC20 sourceTokenContract = IERC20(sourceToken);
+
+			uint256 balanceBefore = sourceTokenContract.balanceOf(address(this));
+
 			IERC20(sourceToken).safeTransferFrom(msg.sender, address(this), sourceTokenAmount);
+
+			// explicit balance check so that we can support deflationary tokens
```

Comment:
Deflationary token design is an anti-pattern.
Personally, I'd vote against their support as a way to discourage people implementing tokens that don't comply with ERC20 standard